src/formatDate.js

```diff
@@ -8,7 +8,68 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  if (typeof date !== 'string' || date.length === 0) {
+    return '';
+  }
+
+  if (
+    !fromFormat ||
+    !toFormat ||
+    typeof fromFormat !== 'object' ||
+    typeof toFormat !== 'object' ||
+    !('length' in fromFormat) ||
+    !('length' in toFormat)
+  ) {
+    return '';
+  }
+
+  const fromSeparator = fromFormat[3];
+  const toSeparator = toFormat[3];
+
+  if (!fromSeparator || !toSeparator) {
+    return '';
+  }
+
+  const parts = date.split(fromSeparator);
+
+  if (parts.length !== 3) {
+    return '';
+  }
+
+  const partsMap = {
+    DD: '',
+    MM: '',
+    YYYY: '',
+    YY: '',
+  };
+
+  fromFormat.slice(0, 3).forEach((part, index) => {
+    if (partsMap.hasOwnProperty(part)) {
+      partsMap[part] = parts[index];
+    }
+  });
```

Comment:
Using `hasOwnProperty` is unnecessary here. Since `partsMap` is defined in the same scope and is not inheriting from any prototype that might have additional properties, you can assign the values directly.